presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -305,6 +302,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     newJoinPredicate = joinPredicate; // Use the same as the original
                     break;
                 case FULL:
+                case CROSS:
```

Comment:
For the CROSS join, i think we should be able push all the *deterministic* predicates down that can be rewritten entirely in terms of each side of the join.